oak_attestation_verification/src/verifier.rs

```diff
@@ -134,3 +166,36 @@ fn verify_cb(
 ) -> anyhow::Result<()> {
     anyhow::bail!("Needs implementation")
 }
+
+fn verify_transparent_release_endorsement(
+    now_utc_millis: i64,
+    endorsement: &TransparentReleaseEndorsement,
+    reference_value: &BinaryReferenceValue,
+) -> anyhow::Result<()> {
+    let statement = parse_endorsement_statement(&endorsement.endorsement)?;
+    let actual = get_digest(&statement)?;
+
+    match reference_value.r#type.as_ref() {
+        Some(binary_reference_value::Type::Endorsement(erv)) => verify_binary_endorsement(
+            now_utc_millis,
+            &endorsement.endorsement,
+            &endorsement.rekor_log_entry,
+            &erv.endorser_public_key,
+            &erv.rekor_public_key,
+        ),
+        Some(binary_reference_value::Type::Digests(ds)) => {
+            for raw in ds.digests.iter() {
+                let expected = raw_to_hex_digest(&raw);
+                let match_result = is_hex_digest_match(&actual, &expected);
+                if match_result == MatchResult::SAME {
+                    return Ok(());
+                }
+                if match_result == MatchResult::DIFFERENT {
+                    anyhow::bail!("found digest deviation");
+                }
+            }
```

Comment:
Perhaps something like

```rust
if ds.digests.iter().any(|actual| digest_match(&actual, &expected)) {
  ...
```

Reply:
We have three states, so this would mean we'd need to iterate several times which seems worse. Unless I am missing something.

Reply:
Still not sure about that tri-state TBH. I think something like implementing a method `is_subset_of` for Digest that returns a bool would be more idiomatic. In fact I am not sure I am reading the code here correctly: there is a list of digests, any of which should be sufficient for a match. But if one of them is found to be `DIFFERENT`, then we return an error immediately? That does not sound right (but I'm probably missing something).